### PR TITLE
[artifactory] Simplify nginx setup

### PR DESCRIPTION
**Pull request by eldada**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated

Now that nginx config is managed in `values.yaml`, we can simplify init setup of the nginx pod.

### CODE REVIEW

File: stable/artifactory/templates/nginx-deployment.yaml [hunk 1]
```diff
@@ -37,37 +37,18 @@ spec:
...
-      - name: "remove-lost-found"
+      - name: "setup"
         image: "{{ .Values.initContainerImage }}"
         imagePullPolicy: {{ .Values.nginx.image.pullPolicy }}
         command:
         - '/bin/sh'
         - '-c'
-        - 'rm -rfv {{ .Values.nginx.persistence.mountPath }}/lost+found'
+        - >
+          [ -d {{ .Values.nginx.persistence.mountPath }}/lost+found ] && rm -rfv {{ .Values.nginx.persistence.mountPath }}/lost+found;
```
Comment (danielezer) [inline stable/artifactory/templates/nginx-deployment.yaml]: @eldada is it really necessary to check I
if the directory exists? mew -rf will be good even when the directory doesn’t exist

File: stable/artifactory/templates/nginx-deployment.yaml [hunk 2]
```diff
@@ -37,37 +37,18 @@ spec:
...
-        volumeMounts:
-          - mountPath: {{ .Values.nginx.persistence.mountPath | quote }}
-            name: nginx-volume
-      - name: "wait-for-artifactory"
-        image: "{{ .Values.initContainerImage }}"
-        command:
-        - 'sh'
-        - '-c'
-        - >
-          until nc -z -w 2 {{ $serviceName }} {{ $servicePort }} && echo artifactory ok; do
```
Comment (danielezer) [inline stable/artifactory/templates/nginx-deployment.yaml]: What about the wait-for-artifactory? 
Wouldn’t this cause a lot of nginx restarts?

Comment (rimusz) [reply stable/artifactory/templates/nginx-deployment.yaml]: I should keep the `wait-for-artifactory`

File: stable/artifactory/templates/nginx-deployment.yaml (hunk 1 above)
Comment (eldada) [reply stable/artifactory/templates/nginx-deployment.yaml]: Agree. Removed.

File: stable/artifactory/templates/nginx-deployment.yaml (hunk 2 above)
Comment (eldada) [reply stable/artifactory/templates/nginx-deployment.yaml]: It was restarting anyway... the wait-for-artifactroy was not doing a good job anyway. I'd rather not use a faulty init container and simplify the setup. 
It's better the pos restarts on liveness than on faulty config.

**Pull request merged**